app/api/v2/handlers/fact_source_api.py

```diff
@@ -22,44 +22,84 @@ def add_routes(self, app: web.Application):
+                          parameters=[{
+                              'in': 'path',
+                              'name': 'id',
+                              'description': 'The id of the fact source',
+                              'schema': {'type': 'string'},
+                              'required': 'true'
+                            }])
     @aiohttp_apispec.querystring_schema(BaseGetOneQuerySchema)
-    @aiohttp_apispec.response_schema(SourceSchema(partial=True))
+    @aiohttp_apispec.response_schema(SourceSchema(partial=True), description='Returns a single Source dumped in SourceSchema format.')
     async def get_fact_source_by_id(self, request: web.Request):
         source = await self.get_object(request)
         return web.json_response(source)
 
+    @aiohttp_apispec.docs(tags=['sources'],
+                              summary='Create a fact source.',
+                              description='Create a new Fact Source using the format provided in the SourceSchema.')
     @aiohttp_apispec.docs(tags=['sources'])
     @aiohttp_apispec.request_schema(SourceSchema)
-    @aiohttp_apispec.response_schema(SourceSchema)
+    @aiohttp_apispec.response_schema(SourceSchema, description='Returns a single Source dumped in SourceSchema format.')
     async def create_fact_source(self, request: web.Request):
         source = await self.create_on_disk_object(request)
         return web.json_response(source.display)
 
-    @aiohttp_apispec.docs(tags=['sources'], summary='All fields can be updated excluding "id" '
-                                                    'and "adjustments".')
+    @aiohttp_apispec.docs(tags=['sources'],
+                          summary='Update an existing fact source.',
+                          description='Returns an updated fact source. All fields in a fact source can be updated, except for "id" and "adjustments".',
+                          parameters=[{
+                            'in': 'path',
+                            'name': 'id',
+                            'description': 'The id of the fact source.',
+                            'schema': {'type': 'string'},
+                            'required': 'true'
+                          }])
     @aiohttp_apispec.request_schema(SourceSchema(partial=True))
-    @aiohttp_apispec.response_schema(SourceSchema)
+    @aiohttp_apispec.response_schema(SourceSchema, description='Returns a single Source dumped in SourceSchema format.')
     async def update_fact_source(self, request: web.Request):
         source = await self.update_on_disk_object(request)
         return web.json_response(source.display)
 
-    @aiohttp_apispec.docs(tags=['sources'])
+    @aiohttp_apispec.docs(tags=['sources'],
+                          summary='Update an existing or create a new fact source.',
+                          description='Returns an updated fact source. All fields in a fact source can be updated, except for "id" and "adjustments".',
```

Comment:
For the description here, I think it might be best to change it to something like "Use fields from the SourceSchema in the request body to replace an existing Fact Source or create a new Fact Source." The PUT request will create a new Fact Source object, and either replace the existing Fact Source corresponding to the ID in the url or it will create a new one, so the request body behaves a little more like POST than PATCH.